internal/git/util.go

```diff
@@ -84,13 +116,16 @@ func GetLatestManifests(
 		// Read all the files under the path and apply each one respectively.
 		err = tree.Files().ForEach(func(f *object.File) error {
 			logger.Debugw("read file", "file_name", f.Name)
-			// TODO: this currently assumes that one file contains just one manifest - modify for multiple
 			manifest, err := f.Contents()
 			if err != nil {
 				logger.Errorw("cannot get file content", "filename", f.Name, "err", err)
 				return err
 			}
-			manifests = append(manifests, manifest)
+			manifestData, err := SplitYAMLToString([]byte(manifest))
+			if err != nil {
+				return fmt.Errorf("can not parse file data, err: %v", err)
+			}
+			manifests = append(manifests, manifestData...)
```

Comment:
This change will fix this issue https://github.com/numaproj-labs/numaplane/issues/128